test/mocha/helpers.js

```diff
@@ -22,11 +24,31 @@ const {Ed25519Signature2018, RsaSignature2018} = suites;
 const sinon = require('sinon');
 const {Ed25519KeyPair} = require('crypto-ld');
 
+const getRandomBytes = promisify(crypto.randomBytes);
+
 // for key generation
 exports.KMS_MODULE = 'ssm-v1';
 // algorithm required for the jwe headers
 exports.JWE_ALG = 'ECDH-ES+A256KW';
 
+const largeRange = Number.MAX_SAFE_INTEGER - Math.pow(2, 32);
+
+// gets a number between MAX_SAFE_INTEGER & 2^32.
+exports.largeNumber = () => {
+  // if Math.random is 1 the number will be 2^32 otherwise it will be larger.
+  return Number.MAX_SAFE_INTEGER - Math.floor(largeRange * Math.random());
```

Comment:
-1 to using randomness here as we will have no idea what actually got tested or what works.

@dlongley will have to say exactly what the need is here, but I would suggest some fixed/deterministic series of test that test min/middle/max values that are supposed to work.

Reply:
we do know what got tested a number larger than `Math.pow(2, 32)` but smaller than `Number.MAX_SAFE_INTEGER`. This also tests all numbers in that range albeit slowly @davidlehn recommended.

Reply:
I agree it should at least test a static set of values around the limits as appropriate.  Something like 2^32-1, 2^32, 2^32+1, (MAX-2^32)/2, MAX-1, MAX.  And maybe toss in some error cases like -1 (and 0?).  I'm assuming logic handling this doesn't know about BigInts yet?

Reply:
I'm willing to add the tests for min `Math.pow(2,32)` middle, and max `Number.MAX_SAFE_INTEGER` asked for provided we still allow the one I already wrote as I do think fuzz testing this functionality has value in the long run and it's not a good idea to waste work. So let me know. I can add the tests in a moment.
I also like @davidlehn idea of adding tests for negative sequence numbers to ensure it errors when that happens although that was not part of the issue.

@mattcollier  @davidlehn do we have a deal? thumbs up if so.

Reply:
Assuming this can be marked resolved taken @davidlehn 's changes? My apologies about the wait.